### ISSUE TITLE
Can't decode some "valid" x.509 certificates

### ISSUE THREAD

**Issue by tsileo** [wbond/asn1crypto#8]
Hi Will,

There's a few valid certificate that asn1crypto fails to parse (whereas OpenSSL is able to decode them).

Here is the traceback:

```
Traceback (most recent call last):
  File "[...]/collect_report/__init__.py", line 181, in _extract_common_name_from_certificate_chain
    subject_fields = tbs_cert['subject'].native
  File "[...]/../lib/asn1crypto/x509.py", line 1015, in native
    for rdn in self.chosen.native:
  File "[...]/../lib/asn1crypto/core.py", line 3304, in native
    self._parse_children(recurse=True)
  File "[...]/../lib/asn1crypto/core.py", line 3276, in _parse_children
    raise e
ValueError: Value [UNIVERSAL 22] did not match the class and tag of any of the alternatives in asn1crypto.x509.DirectoryString: [UNIVERSAL 12], [UNIVERSAL 30], [UNIVERSAL 19], [UNIVERSAL 28], [UNIVERSAL 20]
    while parsing asn1crypto.x509.NameTypeAndValue
    while parsing asn1crypto.x509.RelativeDistinguishedName
    while parsing asn1crypto.x509.RDNSequence
```

I've tried to debug the issue myself, but I'm stuck, it would be nice if you can help.

Can I send you an email with two x509 certificates that asn1crypto fails to parse?

Thanks!

**Comment by wbond** [comment]
Yes, please do send them to will at wbond.net. Some X.509 certs don't follow the spec and I have to add special parsing for how they are constructed.

**Comment by tsileo** [comment]
Thanks! I sent you an email with the certificates.

**Comment by wbond** [comment]
@tsileo Would it be ok if I add one of these certificates to the test suite? It would be publicly available via GitHub if I did that. I understand if you've prefer not to.

The issue these aren't being parsed properly is because they use the IA5String ASN.1 encoding in a relative distinguished name value that only allows either PrintableString, UTF8String, BMPString, UniversalString or TeletexString. While technically invalid, I'd rather have them parse since I imagine other software is parsing these also.

**Comment by wbond** [comment]
The fix for this is part of version 0.17.0.
